File: src/python/src/de/yaxgl/Controls/ImageBox.py
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;

namespace de.yaxgl
{
    public class ImageBox : Control
    {
        public ImageBox(Container owner, string ID)
        {
            this.owner = owner;
            this.ID = ID;

            this.control = new System.Windows.Forms.PictureBox();
            //events
            this.control.Click += new System.EventHandler(clickEvent);
        }

        public override void initializeNativeControl(System.Xml.XmlElement xmlElement)
        {
            setBounds(Convert.ToInt32(xmlElement.Attributes["xpos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["ypos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["width"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["height"].InnerText));
            loadImage(xmlElement.Attributes["source"].InnerText);
        }
        
        public void loadImage(string fileImage)
        {
            ((System.Windows.Forms.PictureBox)this.control).Load(Application.StartupPath + "/" + fileImage);
        }

    }
}

File: trunk/src/gwt/yaxgl-gwt-lib/src/de/yaxgl/client/Controls/ImageBox.java
package de.yaxgl.client.Controls;



import com.google.gwt.xml.client.Element;

import de.yaxgl.client.Base.Control;
import de.yaxgl.client.Container.Container;

public class ImageBox extends Control {

	public ImageBox(Container owner,String ID)
	{
		this.owner=owner;
		this.ID=ID;
		this.control = new com.google.gwt.user.client.ui.Image();
		
		control.addStyleName("yaxgl-imagebox");
		
		this.owner.add(this);
	}
	

	public void initializeNativeControl(Element xmlElement) {
		
		setBounds(Integer.valueOf(xmlElement.getAttribute("xpos")),
				Integer.valueOf(xmlElement.getAttribute("ypos")),
				Integer.valueOf(xmlElement.getAttribute("width")),
				Integer.valueOf(xmlElement.getAttribute("height")));
		
		loadImage(xmlElement.getAttribute("source"));
		
	}
	
	public void loadImage(String fileImage)
    {
		((com.google.gwt.user.client.ui.Image)this.control).setUrl(fileImage);
    }
}
